builtins.go

```diff
@@ -1425,6 +1426,170 @@ func builtinParseYAML(i *interpreter, str value) (value, error) {
 	return jsonToValue(i, elems[0])
 }
 
+func builtinParseCSVWithHeader(i *interpreter, arguments []value) (value, error) {
+	strv := arguments[0]
+	dv := arguments[1]
+
+	sval, err := i.getString(strv)
+	if err != nil {
+		return nil, err
+	}
+	s := sval.getGoString()
+
+	d := ',' // default delimiter
+	if dv.getType() != nullType {
+		dval, err := i.getString(dv)
+		if err != nil {
+			return nil, err
+		}
+		ds := dval.getGoString()
+		if len(ds) != 1 {
+			return nil, i.Error(fmt.Sprintf("Delimiter %s is invalid", ds))
+		}
+		d = rune(ds[0]) // conversion to rune
+	}
+
+	json := make([]interface{}, 0)
+	var keys []string
+
+	reader := csv.NewReader(strings.NewReader(s))
+	reader.Comma = d
+
+	for row := 0; ; row++ {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, i.Error(fmt.Sprintf("failed to parse CSV: %s", err.Error()))
+		}
+
+		if row == 0 { // consider first row as header
+			// detect and handle duplicate headers
+			keyCount := map[string]int{}
+			for _, k := range record {
+				keyCount[k]++
+				if c := keyCount[k]; c > 1 {
+					keys = append(keys, fmt.Sprintf("%s__%d", k, c-1))
```

Comment:
I think this is actually going to be confusing because it's not going to round-trip back through the manifest function (the mangled column names will appear in the output).  Also the behaviour would have to be documented which would be a pain.  Now you've renamed the function and it doesn't look like a generic CSV reader anymore I think what you were originally doing was fine.

Reply:
I guess we can add an argument called `handle_duplicate_headers=false` for users to opt for this. Using this flag, we could either overwrite the duplicate headers or rename the duplicate fields as above. WDYT?

Reply:
SGTM

Reply:
Made the changes